File: post/src/main/java/com/example/learningwebflux/post/platformconnection/PlatformConnection.java
package com.example.learningwebflux.post.platformconnection;

import com.example.learningwebflux.post.Platform;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

@Table("platform_connection")
public class PlatformConnection {
    public PlatformConnection(String id, String userId, Platform platform, String platformUserId, String accessToken) {
        this.id = id;
        this.userId = userId;
        this.platform = platform;
        this.platformUserId = platformUserId;
        this.accessToken = accessToken;
    }

    @Id
    public final String id;
    public final String userId;
    public final Platform platform;
    public final String platformUserId;
    public String accessToken;
}

File: postanalytics/src/main/kotlin/com/example/learningwebflux/postanalytics/platformconnection/PlatformConnection.kt
package com.example.learningwebflux.postanalytics.platformconnection

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table

@Table
internal class PlatformConnection(
    @Id
    public val id: String,
    public val userId: String,
    public val platformUserId: String,
    public var accessToken: String?,
)